### DISCUSSION TITLE
PiVPN on Linux Mint

### DISCUSSION THREAD

**Discussion by timboxyz** [pivpn/pivpn#1802] [Show and tell]
Since PiVPN can be made to work on Ubuntu I reasoned that it should be possible to get it running on Linux Mint as an Ubuntu derivative. As I had an old netbook with that OS on I wasn't otherwise using I thought I would give it a try.

As the installer stands it doesn't recognise Mint as a valid platform but a couple of small tweaks fixed that and it now runs quite happily.
I am running Mint 21.2 (Victoria) a jammy derivative and made the following changes to` install.pivpn.io`

In `distroCheck()` I added the following (around line 364 with before and after lines shown for context)

```
    # Alpine support
    if [[ -z "${OSCN}" ]]; then
      OSCN="${VER}"
    fi
  fi

# Workaround for Linux Mint Victoria
# assumes results from lsb_release
  [[ "${PLAT}" = "Linuxmint" ]] && PLAT="Ubuntu"

  [[ "${OSCN}" = "victoria" ]] && OSCN="jammy"
# end workaround

  case "${PLAT}" in
    Debian | Raspbian | Ubuntu)
      case "${OSCN}" in

```
It relies upon `lsb_release` being installed and the results from that being used rather than the fall back of inspecting `/etc/os-release`. Since that was the case for my install I didn't bother with hacking the fall back code too.

**Comment by coolapso** [comment]
Mint is not a OS Destined to run as a server therefore not supported intentionally.

**Reply by timboxyz** [reply]
Quite, but it made an easy test bed to try things out on so i thought why not give it a spin as is rather than re-OS just for a try out.

As it worked OK I thought others might be interested in a low impact play pen before committing to a dedicated build.

**Reply by mengagumkan** [reply]
Thanks timboxez. 
I applied your code and pivpn works perfectly on Mint 21.2 Mate laptop OS.
First I installed wireguard and wireguard-tools from Software Manager.
Then I downloaded the script install.ipvpn.io from internet manually, copied it to Text Editor  and inserted the code that you proposed in the right place. Then saved the file in my home root directory renaming the file to install.pivpn.sh. Then I run the code in terminal:

sudo chmod +x install.pivpn.sh

Then I run the command:

sudo curl -L "file:///home/XXX/install.pivpn.sh" | bash

From this point on I successfuly installed pivpn and now it works like a breeze.

Did I mention that a few days before, I successfully installed pi-hole on the same laptop and it works perfectly? For that I applied the command:

sudo curl -sSL https://install.pi-hole.net | PIHOLE_SKIP_OS_CHECK=true sudo -E bash

**Reply by mengagumkan** [reply]
I just thought that someone might use the corrected installation file:

[install.pivpn.sh.zip](https://github.com/user-attachments/files/16098063/install.pivpn.sh.zip)
